From stable-bounces@linux.kernel.org Thu Apr 27 05:23:14 2006
Message-Id: <20060427122214.159808000@linux-mips.org>
Date: Thu, 27 Apr 2006 00:00:01 +0100
From: Ralf Baechle <ralf@linux-mips.org>
To: stable@kernel.org
Content-Disposition: inline; filename=0001.patch
Cc: 
Subject: MIPS: Use "R" constraint for cache_op.

Gcc might emit an absolute address for the the "m" constraint which
gas unfortunately does not permit.
    
Signed-off-by: Ralf Baechle <ralf@linux-mips.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 include/asm-mips/r4kcache.h |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- linux-2.6.16.11.orig/include/asm-mips/r4kcache.h
+++ linux-2.6.16.11/include/asm-mips/r4kcache.h
@@ -37,7 +37,7 @@
 	"	cache	%0, %1					\n"	\
 	"	.set	pop					\n"	\
 	:								\
-	: "i" (op), "m" (*(unsigned char *)(addr)))
+	: "i" (op), "R" (*(unsigned char *)(addr)))
 
 static inline void flush_icache_line_indexed(unsigned long addr)
 {
